unittests/test_policies.py

```diff
@@ -257,6 +261,30 @@ def test_dependencies(self):
             if t.ref_count == 0:
                 assert os.path.exists(os.path.join(check.outputdir, 'out.txt'))
 
+    def test_sigterm(self):
+        # Wrapper of self.runall which is used from a child process and
+        # passes the thrown exception to the parent process
+        def _runall(checks, conn):
+            ex = None
+            try:
+                self.runall(checks)
+            except ReframeForceExitError as e:
+                ex = e
+
+            conn.send(ex)
+            conn.close()
+
+        with pytest.raises(ReframeForceExitError, match='Received SIGTERM'):
+            parent_conn, child_conn = Pipe(False)
```

Comment:
Is there a way to use a `Pipe` with a `with` statement? According to the documentation, `Connection` objects are context managers.

Reply:
```suggestion
            rd_endpoint, wr_endpoint = Pipe(duplex=False)
```